process/dataindexer/dataIndexer_test.go

```diff
@@ -58,19 +48,29 @@ func TestDataIndexer_NewIndexerWithCorrectParamsShouldWork(t *testing.T) {
 }
 
 func TestDataIndexer_SaveBlock(t *testing.T) {
-	called := false
+	count := 0
 
 	arguments := NewDataIndexerArguments()
-	arguments.DataDispatcher = &mock.DispatcherMock{
-		AddCalled: func(item workItems.WorkItemHandler) {
-			called = true
-		},
-	}
 	arguments.BlockContainer = &mock.BlockContainerStub{
 		GetCalled: func(headerType core.HeaderType) (dataBlock.EmptyBlockCreator, error) {
 			return dataBlock.NewEmptyHeaderV2Creator(), nil
 		},
 	}
+
+	arguments.ElasticProcessor = &mock.ElasticProcessorStub{
+		SaveHeaderCalled: func(outportBlockWithHeader *outport.OutportBlockWithHeader) error {
+			count++
```

Comment:
might define a map of counters, since for example `SaveHeaderCalled` could have been called twice, and `SaveMiniblockCalled` not even once, but the test still passes

Reply:
done